plugins/community/neoeinstein-prost/v0.2.3/Dockerfile

```
@@ -0,0 +1,12 @@
+# syntax=docker/dockerfile:1.4
+FROM rust:1.70.0-alpine3.18 as builder
+RUN apk add --no-cache musl-dev
+WORKDIR /app
+ENV CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse
+RUN --mount=type=cache,target=/usr/local/cargo/registry,sharing=locked --mount=type=cache,target=/root/target \
```

```suggestion
RUN --mount=type=cache,target=/usr/local/cargo/registry,sharing=locked --mount=type=cache,target=/root/target \
```

Added `sharing=locked` as apparently it isn't safe to have multiple writers to the Cargo registry (maybe a new regression in newer versions of Rust?).

---

I guess this makes sense, at first I thought we could `sharing=private` so that each build creates a cached directory, but we want them to be re-used, just not in parallel?

---

Favored this way to avoid multiple downloads of the same content, with the thought that they'd likely pull in a lot of the same dependencies.